packages/paste-core/components/alert/src/index.tsx

```diff
@@ -0,0 +1,126 @@
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {Button} from '@twilio-paste/button';
+import {CloseIcon} from '@twilio-paste/icons/esm/CloseIcon';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
+import {InformationIcon} from '@twilio-paste/icons/esm/InformationIcon';
+import {WarningIcon} from '@twilio-paste/icons/esm/WarningIcon';
+
+export const ROLES = {
+  error: 'alert',
+  neutral: 'status',
+  warning: 'alert',
+} as const;
+export const VARIANTS = {
+  error: 'error',
+  neutral: 'neutral',
+  warning: 'warning',
+} as const;
+export const BACKGROUND_COLORS = {
+  error: 'colorBackgroundErrorLightest',
+  neutral: 'colorBackgroundNeutralLightest',
+  warning: 'colorBackgroundWarningLightest',
+} as const;
+export const BORDER_COLORS = {
+  error: 'colorBorderErrorLight',
+  neutral: 'colorBorderNeutralLight',
+  warning: 'colorBorderWarningLight',
+} as const;
+
+export type Variants = ValueOf<typeof VARIANTS>;
+export type BackgroundColors = ValueOf<typeof BACKGROUND_COLORS>;
+export type BorderColors = ValueOf<typeof BORDER_COLORS>;
+export type Roles = ValueOf<typeof ROLES>;
+
+export interface AlertProps {
+  id?: never;
+  className?: never;
+  children: NonNullable<React.ReactNode>;
+  onDismiss?: () => void;
+  role?: string;
+  variant: Variants;
+}
+
+const getBackgroundColor = (variant: Variants): BackgroundColors => {
+  switch (variant) {
+    case VARIANTS.error:
+      return BACKGROUND_COLORS.error;
+    case VARIANTS.warning:
+      return BACKGROUND_COLORS.warning;
+    case VARIANTS.neutral:
+    default:
+      return BACKGROUND_COLORS.neutral;
+  }
+};
+
+const getBorderColor = (variant: Variants): BorderColors => {
+  switch (variant) {
+    case VARIANTS.error:
+      return BORDER_COLORS.error;
+    case VARIANTS.warning:
+      return BORDER_COLORS.warning;
+    case VARIANTS.neutral:
+    default:
+      return BORDER_COLORS.neutral;
+  }
+};
+
+const renderAlertIcon = (variant: Variants): React.ReactElement => {
+  switch (variant) {
+    case VARIANTS.error:
+      return <ErrorIcon iconColor="colorTextErrorDark" decorative={false} title="error: " size="sizeIcon30" />;
+    case VARIANTS.warning:
+      return <WarningIcon iconColor="colorTextWarningDark" decorative={false} title="warning: " size="sizeIcon30" />;
+    case VARIANTS.neutral:
+    default:
+      return <InformationIcon iconColor="colorTextIcon" decorative={false} title="information: " size="sizeIcon30" />;
+  }
+};
+
+const getComputedRole = (variant: Variants, role: string | undefined): string | Roles => {
+  if (role != null) return role;
+  if (variant === VARIANTS.error || variant === VARIANTS.warning) {
+    return ROLES.error;
+  }
+  return ROLES.neutral;
+};
+
+const Alert: React.FC<AlertProps> = ({children, onDismiss, variant, role, ...props}) => {
+  return (
+    <Box
+      {...safelySpreadBoxProps(props)}
+      backgroundColor={getBackgroundColor(variant)}
```

Comment:
Since we share the same keys among all the objects defined above on lines 12-31, we can do something like this and remove all these functions/switches

```js
backgroundColor={BACKGROUND_COLORS[variant]}
borderColor={BORDER_COLORS[variant]}
role={role != null ? role : ROLES[variant]}
```
Typescript should make this safe to do, but additionally we can define the objects like this
```js
const BackgroundColors = {
 [ERROR]: 'colorBackgroundErrorLightest'
  ...
}
```
in JS land to make sure the keys matched.  We don't need that here though

Reply:
oooh yeah, that's much better